lib/Sema/CSDiagnostics.cpp

```diff
@@ -6224,3 +6224,23 @@ bool CoercionAsForceCastFailure::diagnoseAsError() {
       .highlight(getSourceRange());
   return true;
 }
+
+bool KeyPathRootTypeMismatchFailure::diagnoseAsError() {
+  auto &cs = getConstraintSystem();
+  auto locator = getLocator();
+  auto anchor = locator->getAnchor();
+
+  assert(locator->isKeyPathRoot() && "Expect a key path root");
+
+  if (auto *subscriptExpr = dyn_cast_or_null<SubscriptExpr>(anchor)) {
+    auto *tupleExpr = cast<TupleExpr>(subscriptExpr->getIndex());
+    auto keyPathType = cs.getType(tupleExpr->getElement(0));
+    auto baseType = cs.getType(subscriptExpr->getBase());
```

Comment:
We want to avoid querying the constraint system during diagnosis, as its state may not be the same as it was when the fix was created. Instead it should be accessed via the solution, which can be done by just dropping the `cs.` :)

Reply:
Fixed :)